hazelcast/src/main/java/com/hazelcast/query/impl/BaseIndexStore.java

```diff
@@ -239,21 +266,39 @@ private Comparable sanitizeScalar(Object input) {
 
     }
 
-    private static class PassThroughFunctor implements CopyFunctor<Data, QueryableEntry> {
+    private class PassThroughFunctor implements CopyFunctor<Data, QueryableEntry> {
 
         @Override
         public Map<Data, QueryableEntry> invoke(Map<Data, QueryableEntry> map) {
+            if (isExpirable()) {
+                long now = Clock.currentTimeMillis();
```

Comment:
WDYT about wrapping the returned map into a special access updating wrapper? In that case we will avoid double iteration while sending results to the client, for instance.

Reply:
Yes, I will do this improvement.